### PR TITLE
PL-38935-JIT-user-provisioning

### PR DESCRIPTION
**Pull request by rashminandasahoo**
Preview - https://64902fd4cc1ae54b4516a21f--harness-developer.netlify.app/docs/platform/User-Management/provision-use-jit
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by VikasMaddukuriHarness — DISMISSED** [review]

File: docs/platform/3_User-Management/provision-use-jit.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+```
+
+:::important
+Currently, this feature is behind the feature flag `PL_ENABLE_JIT_USER_PROVISION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+:::
+
+Just-in-time (JIT) provisioning in Harness lets you provision users automatically upon first sign in to Harness through SAML SSO.
+You can automatically provision users added to your SAML applications configured in Harness and allow them to access Harness.
```
Comment (StrawberryPad) [inline docs/platform/3_User-Management/provision-use-jit.md:17]: I can't tell if it's the users that are configured in Harness, or the SAML apps that are configured in Harness. Also "them" is unclear... though I'm guessing it's the users.

File: docs/platform/3_User-Management/provision-use-jit.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Provisioning users automatically eliminates repetitive tasks related to manual provisioning, simplifying user management.
+
+:::important
+Harness supports Just-in-time provisioning only if a new user attempts to log in via IDP.
```
Comment (StrawberryPad) [inline docs/platform/3_User-Management/provision-use-jit.md:21]:
```suggestion
Harness supports JIT provisioning only if a new user attempts to log in via IDP.
```

File: docs/platform/3_User-Management/provision-use-jit.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+   ```
+
+5. Select **Enable JIT Provisioning**. 
+   This enables Just-In-Time (JIT) provisioning to add users to Harness on their first login via the configured SAML provider.
+
+6. To control who can get added to Harness on their first login, define the key-value that must be present in the SAML assertions by entering the following: 
+   - **JIT Validation Key**: To log in for the first time, the SAML assertion must contain the key.
+   - **JIT Validation Value**: At the time of first login, enter the value for the validation key that must be present in the SAML assertion.
+   
+     ![](./static/jit-user-provisioning.png)
```
Comment (StrawberryPad) [inline docs/platform/3_User-Management/provision-use-jit.md:77]: I think our convention is to have the image you might be referring to before the instruction on what to do. So if following this, this image should be in step 5.

File: docs/platform/3_User-Management/provision-use-jit.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+1. Select **ACCOUNT SETTINGS**, and then **Authentication**.
+2. Select **SAML Provider**.
+   The **Add SAML Provider** settings appear.
+3. In **Name**, enter a name for the SAML provider.
+4. In **Select a SAML Provider**, select one of the following: 
+   - Azure
+   - Okta
+   - OneLogin
+   - Other
+
```
Comment (StrawberryPad) [inline docs/platform/3_User-Management/provision-use-jit.md:36]: This is probably not a good use of tabs. You can simply add the link to instructions to the bullet list above. This is easier for the user, and more importantly here reduces the cost of maintenance of this doc (having tabs unnecessarily adds to maintenance time... important to consider when we're very short on resources).

File: docs/platform/3_User-Management/provision-use-jit.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+   ```mdx-code-block
+      </TabItem>
+   </Tabs>
+   ```
+
+5. Select **Enable JIT Provisioning**. 
+   This enables Just-In-Time (JIT) provisioning to add users to Harness on their first login via the configured SAML provider.
+
+6. To control who can get added to Harness on their first login, define the key-value that must be present in the SAML assertions by entering the following: 
+   - **JIT Validation Key**: To log in for the first time, the SAML assertion must contain the key.
```
Comment (StrawberryPad) [inline docs/platform/3_User-Management/provision-use-jit.md:74]:
```suggestion
   - **JIT Validation Key**: The SAML assertion must contain this key on first login.
```

File: docs/platform/3_User-Management/provision-use-jit.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+   - **JIT Validation Value**: At the time of first login, enter the value for the validation key that must be present in the SAML assertion.
```
Comment (StrawberryPad) [inline docs/platform/3_User-Management/provision-use-jit.md:75]:
```suggestion
   - **JIT Validation Value**: The value for the validation key that must be present in the SAML assertion on first login.
```

**Review by StrawberryPad — CHANGES_REQUESTED** [review]
Requesting some changes for clarity.

File: docs/platform/3_User-Management/provision-use-jit.md (hunk 1 above)
Comment (rashminandasahoo) [reply docs/platform/3_User-Management/provision-use-jit.md:17]: Rephrased.

File: docs/platform/3_User-Management/provision-use-jit.md (hunk 3 above)
Comment (rashminandasahoo) [reply docs/platform/3_User-Management/provision-use-jit.md:77]: Done.

File: docs/platform/3_User-Management/provision-use-jit.md (hunk 4 above)
Comment (rashminandasahoo) [reply docs/platform/3_User-Management/provision-use-jit.md:36]: Done.

**Pull request merged**